src/components/ListItem.jsx

```diff
@@ -1,5 +1,45 @@
+import { useEffect, useState } from 'react';
+import { updateItem } from '../api/firebase';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils';
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ listPath, item, name }) {
+	const [isChecked, setIsChecked] = useState(false);
+
+	function handleChange(e) {
+		// ensures checked box can be toggled
+		setIsChecked(!isChecked);
+		// pass listPath, copies the item object and sets a new value named isChecked
+		updateItem(listPath, { ...item, isChecked: !isChecked });
+	}
+
+	// this useEffect will run everytime the value of "dateLastPurchased" on the item object changes
+	useEffect(() => {
+		//
+		if (item.dateLastPurchased) {
+			const now = new Date();
+			// calculate expiration time by adding on 24 hours onto the value that is stored on the dateLastPurchased key
+			// this variable will be in milliseconds
+			const expirationTime =
+				new Date(
+					item.dateLastPurchased.seconds * 1000 +
+						Math.floor(item.dateLastPurchased.nanoseconds / 1000000),
```

Comment:
To make this more readable and reusable we can move this to the function called `getMilliseconds` in utils
```
export function getMilliseconds(date) {
   return date.seconds * 1000 + Math.floor(date.nanoseconds / 1000000)
}
```

and then, we can use it here and further if needed like this:
```
const millisecondsLastPurchased = getMilliseconds(item.dateLastPurchased);
const expirationTime = new Date(millisecondsLastPurchased).getTime()
```

Reply:
Thanks for the suggestion, we refactored this portion slightly for readability.